mpqp/execution/providers/ibm.py

```diff
@@ -651,11 +651,17 @@ def extract_result(
             elif job.job_type == JobType.SAMPLE:
                 assert job.measure is not None
                 counts = result.get_counts(0)
+                job_data = result.data()
                 data = [
                     Sample(
                         bin_str=item,
                         count=counts[item],
                         nb_qubits=job.circuit.nb_qubits,
+                        probability=(
```

Comment:
You dont' need to fill the probability argument if you put the counts and the number of qubits, this will be done directly in the Sample constructor.

Update: I just saw that you modified the way we create samples, and allowed to directly set both if we have them. So you can ignore my remark. The only small problem we can have, is if the counts and the probabilities are not coherent (when retrieved from the provider), we will have sample that have inconsistency between counts and probas (and here recomputing one from the other will avoid that).

Reply:
Yes, that's exactly why I created this feature. We can have the probability we should have and the result we get. I don't think it's a problem to have inconsistencies, is it?

Reply:
If when I print the samples from a result, and I have counts and probability that are not coherent, I think that indeed it is a problem.
I would say that if we allow to input both counts and probas in the creation of sample, we have to check that both are consistent (because we supposed we cannot trust the result coming from the providers), and raise an error if they are not.
But this check will be equivalent to recompute the probas from the counts, or the counts from the probas, so that is why I think at the end that we have to chose one over both inputs (I will go with counts to ensure exactitude of the result, compared to probas that can be rounded).